### DISCUSSION TITLE
Consolidate the unit tests

### DISCUSSION THREAD

**Discussion by gliljas** [nhibernate/nhibernate-core#1409] [General]
The test project contains a lot of NHSpecific unit tests, which have been added, as per the guidelines. After the "asyncification", most of them (?) have been duplicated, adding even more execution time to the test run.

I think it's time to go through them and move them to their respective functionality areas, if they're not already covered by such tests, in which case they can be deleted.

Perhaps it would also be an idea to change the guidelines, so that tests are put in the correct place from the very beginning. With git, GitHub and the PR/code review process, there's not much point in having contributed unit tests in a special structure.

**Comment by fredericDelaporte** [comment]
Maybe could we generate most of the async tests with a specific test category and use it to run them only on the main build. But that would cause async counterparts of tests specific to some db other than SQL Server to never run.
